test/recipes/15-test_ec_comp.t

```diff
@@ -0,0 +1,20 @@
+#! /usr/bin/env perl
+# Copyright 2015-2020 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use File::Spec;
+use OpenSSL::Test qw/:DEFAULT srctop_file/;
+use OpenSSL::Test::Utils;
+
+setup("test_ec_comp");
+
+plan tests => 1;
+
+ok(run(test(["ec_comp_test"])), "running ec_comp_test");
```

Comment:
This should take care of the API usage.

next we should test the CLI apps: we should augment the recipe to also run all the equivalent combinations piping `openssl ec`, `openssl pkey`, etc. either deserializing from the files created with the `generate` script you wrote or generating a new key/params on the fly.
It should be very similar to the loops written in the C test, but replacing function calls with the equivalent cmdline tool.